src/components/map/ExploreLocationMap.tsx

```diff
@@ -0,0 +1,133 @@
+import {Feature} from 'geojson';
+import React, {useCallback, useMemo, useRef, useState} from 'react';
+import {View} from 'react-native';
+import {MapCameraConfig, MapViewConfig} from './MapConfig';
+import {SelectionPin} from './components/SelectionPin';
+import {LocationBar} from './components/LocationBar';
+import {PositionArrow} from './components/PositionArrow';
+import {useControlPositionsStyle} from './hooks/use-control-styles';
+import {SelectionLocationCallback} from './types';
+
+import {isFeaturePoint} from './utils';
+import {Location} from '@atb/favorites';
+import {useFeatureTogglesContext} from '@atb/modules/feature-toggles';
+import {useMapboxJsonStyle} from './hooks/use-mapbox-json-style';
+import {NationalStopRegistryFeatures} from './components/national-stop-registry-features';
+
+type ExploreLocationMapProps = {
+  initialLocation?: Location;
+  onLocationSelect: SelectionLocationCallback;
+};
+
+export const ExploreLocationMap = ({
+  initialLocation,
+  onLocationSelect,
+}: ExploreLocationMapProps) => {
+  const {isMapV2Enabled} = useFeatureTogglesContext();
+  const {getCurrentCoordinates} = useGeolocationContext();
+  const mapCameraRef = useRef<MapboxGL.Camera>(null);
+  const styles = useMapStyles();
+  const controlStyles = useControlPositionsStyle(true);
+
+  const startingCoordinates = useMemo(
+    () =>
+      initialLocation && initialLocation?.resultType !== 'geolocation'
+        ? initialLocation.coordinates
+        : getCurrentCoordinatesGlobal() || FOCUS_ORIGIN,
+    [initialLocation],
+  );
+
+  const [selectedCoordinates, setSelectedCoordinates] =
+    useState(startingCoordinates);
+
+  const onFeatureClick = useCallback(async (feature: Feature) => {
+    if (!isFeaturePoint(feature)) return;
+
+    const {coordinates: positionClicked} = feature.geometry;
+    setSelectedCoordinates({
+      longitude: positionClicked[0],
+      latitude: positionClicked[1],
+    });
+  }, []);
+
+  const onPositionArrowClick = async () => {
+    const coordinates = await getCurrentCoordinates(true);
+    if (coordinates) {
+      mapCameraRef.current?.flyTo(
+        [coordinates.longitude, coordinates.latitude],
+        200,
+      );
+    }
+  };
+
+  const mapboxJsonStyle = useMapboxJsonStyle(false);
+
+  const customMapViewConfigProps = isMapV2Enabled
+    ? {styleURL: undefined, styleJSON: mapboxJsonStyle}
+    : {};
+
+  return (
+    <View style={styles.container}>
+      <LocationBar
+        coordinates={selectedCoordinates || startingCoordinates}
+        onSelect={onLocationSelect}
+      />
+      <View style={{flex: 1}}>
+        <MapboxGL.MapView
+          style={{
+            flex: 1,
+          }}
+          pitchEnabled={false}
+          onPress={onFeatureClick}
+          testID="exploreLocationMapView"
+          {...{
+            ...MapViewConfig,
+            // only updating Map.tsx for now.
+            ...customMapViewConfigProps,
+          }}
```

Comment:
I found this a bit confusing 🤔 Can we write it like so? (Also not sure if the comment still applies)

```suggestion
          {...MapViewConfig}
          {...customMapViewConfigProps}
```

Reply:
This one is also updated in the [next PR](https://github.com/AtB-AS/mittatb-app/pull/5077/files/e1503d5bfeec43704e5fd8df115ac96b05fdc75d..6763b12e3e565033b9c7d0c61c68145294abfbe0#diff-5a86c4dada4a01870eb25bab183849195acfbc143f858c731ceef81d9d102bbd), so merging this as is first.